lib/std/build.zig

```diff
@@ -2555,6 +2555,12 @@ pub const LibExeObjStep = struct {
                 if (cross.cpu.model != std.Target.Cpu.baseline(cross.cpu.arch).model) {
                     try zig_args.append("-mcpu");
                     try zig_args.append(cross.cpu.model.name);
+                } else {
+                    // baseline must be passed for a compilation for binray compatible
+                    // ie: the compiling plateform may have more CPU features than target
+                    // use then baseline
+                    try zig_args.append("-mcpu");
+                    try zig_args.append("baseline");
```

Comment:
Instead of adding this case, why don't we just remove the check if the cpu is the baseline cpu above? The comment should be updated as well to say that even if this is the baseline cpu, the target cpu could be native and we still need to pass `-mcpu`.

Reply:
makes sense for me, parameter is always passed (déterminism) and specify the target (baseline or not)